### PR TITLE
Regression test coverage for AzureML runs

### PR DESCRIPTION
**Pull request by ant0nsc**
Adding the capability for any run in AzureML to compare its results against a set of baseline results.

Please follow the guidelines for PRs contained [here](docs/pull_requests.md). Checklist:

- [x] Ensure that your PR is small, and implements one change.
- [x] Add unit tests for all functions that you introduced or modified.
- [x] Run PyCharm's code cleanup tools on your Python files.
- [x] Link the correct GitHub issue for tracking.
- [x] Update the [Changelog](CHANGELOG.md) file: Describe your change in terms of 
Added/Changed/Removed/... in the "Upcoming" section.
- [x] When merging your PR, replace the default merge message with a description of your PR,
and if needed a motivation why that change was required.

### CODE REVIEW

File: InnerEye/ML/deep_learning_config.py [hunk 1]
```diff
@@ -230,6 +230,13 @@ class WorkflowParams(param.Parameterized):
     monitoring_interval_seconds: int = param.Integer(0, doc="Seconds delay between logging GPU/CPU resource "
                                                             "statistics. If 0 or less, do not log any resource "
                                                             "statistics.")
+    regression_test_folder: Optional[Path] = \
```
Comment (dumbledad) [inline InnerEye/ML/deep_learning_config.py:233]: It's odd to have test related stuff in the non-test code. I'm guessing that's essential?

Comment (ant0nsc) [reply InnerEye/ML/deep_learning_config.py:233]: It is essential here, yes.

File: InnerEye/ML/deep_learning_config.py [hunk 2]
```diff
@@ -230,6 +230,13 @@ class WorkflowParams(param.Parameterized):
...
+        param.ClassSelector(class_=Path, default=None, allow_None=True,
+                            doc="A path to a folder that contains a set of files. At the end of training and "
```
Comment (melanibe) [inline InnerEye/ML/deep_learning_config.py:235]: Should we start the doc by "For testing purposes only"?

**Review by melanibe — APPROVED** [review]

File: InnerEye/ML/baselines_util.py [hunk 3]
```diff
@@ -173,3 +184,119 @@ def get_comparison_baselines(outputs_folder: Path, azure_config: AzureConfig,
...
+        actual_file: Optional[Path] = None
+        if actual_folder:
+            actual_file = actual_folder / file_relative
+            if not actual_file.is_file():
+                actual_file = None
+        elif temp_folder is not None and run is not None:
+            if file_relative in files_in_run:
+                actual_file = temp_folder / file_relative
+                run.download_file(name=str(file_relative), output_file_path=str(actual_file))
+        message = compare_files(expected=file, actual=actual_file) if actual_file else "Missing file"
```
Comment (Shruthi42) [inline InnerEye/ML/baselines_util.py:266]: This should be if actual_file.exists()

**Review by JonathanTripp — APPROVED** [review]

File: InnerEye/ML/baselines_util.py (hunk 3 above)
Comment (ant0nsc) [reply InnerEye/ML/baselines_util.py:266]: In the case of downloading from runs, run.download ensures that the file exists. In the other case, we do the check in line 260, and set it to null if it does not exist. Agreed, it's a bit twisted

File: InnerEye/ML/deep_learning_config.py (hunk 2 above)
Comment (ant0nsc) [reply InnerEye/ML/deep_learning_config.py:235]: not sure it's worth it, the field is already called "regression test"...

**Pull request merged**